src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,46 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final String DEFAULT_COLOR = "WHITE";
+    private static final int MAX_RANGE = 20;
+    private static final int DEFAULT_RADIUS = 10;
+    private Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int randomNameFigure = new Random().nextInt(FiguresName.values().length);
+        switch (FiguresName.values()[randomNameFigure]) {
+            case CIRCLE:
+                Figure circle = new Circle(colorSupplier.getRandomColor(),
+                        random.nextInt(MAX_RANGE) + 1);
+                return circle;
```

Comment:
Don't pass expressions to the constructor. Also, inline the variable.
```suggestion
                int radius = ...;
                return new Circle(color, radius);
```